articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,331 @@
+
+3. #### Do-While Loop
+
+Do-While Loop is an example of [Indefinite Loops](#indefinite-loops).
+
+>Syntax of Do-While Loop.
+
+```Dart
+
+void main(){
+   initialize counter variable;
+   do{
+      //code;
+      increment/decrement;
+   }while(condition);
+
+}
+
+```
+
+>How the Do-While Loop works.
+
+The flow of the Do-While Loop: 
+
+**Code Execution - Increment - Condition Check**.
+
+In Do-While Loop the code executes then the counter variable initialized outside the syntax increments or decrements. Then the condition check, if it's met, it begins from code execution. The code execution will continue until the condition check turns false.
+
+***Example:***
+
+```Dart
+
+void main(){
+   int i = 0;
+
+   do{
+      print ("Hello");
+      i++;
+  }while (i<5);
+
+}
+
+/** Output : Hello
+             Hello
+             Hello
+             Hello
+             Hello **/
+
+```
+The code first executes then the counter variable increments by one. It then checks the condition (1<5). This is true, so the loop will begin again. The code executes, then the counter variable increments by one. It then checks the condition (2<5). This is true, so the loop starts again. This will continue until the code executes and the new counter variable is 5 and the condition checked is (5<5) which is false, so the loop ends after printing 5 hellos.
+
+
+### Other Key Concepts
+
+- #### Break Keyword
+
+We use the `Break Keyword` to 'forcefully' end a loop without changing the main loop statement. It comes in handy when you want a partial output and not to display all the elements.
+
+***Example:***
+
+```Dart
+void main(){
+
+for (int i = 1; i<=10; i++){
+   print ("Hello");
+
+   if(i >= 6){
+      break;
+   }
+
+}
+   
+}
+
+/** Output : Hello
+             Hello
+             Hello
+             Hello
+             Hello **/
+
+```
+This code will end the loop after printing five hellos because of the break statement.
+
+- #### Continue Keyword
+
+We use the `Continue Keyword` to skip code or a loop iteration after the continuation and proceed with the rest of the iteration.
+
+***Example:***
+
+```Dart
+void main(){
+
+for (int i = 1; i<=8; i++){
+   print (i);
+
+   if(i == 5){
+      continue;
+   }
+
+}
+   
+}
+
+/** Output : 1
+             2
+             3
+             4
+             6
+             7
+             8 **/
+
+```
+In the example, when i becomes 5, it skips the print statement, the rest of the loop continues.
+
+
+### Applications of Loops
+
+Here are some positive and negative applications of loops:
+
+1. #### Writing a Music Player Code.
+
+We base a music player on a loop program the enables it to play from one song to the next systematically.
```

Comment:
You can rewrite this as 

`A music player has a loop the enables it to play from one song to the next systematically.`